### DISCUSSION TITLE
Error when trying to install dependencies.

### DISCUSSION THREAD

**Discussion by NetworkJayJay007** [ThioJoe/YT-Spammer-Purge#208] [Q&A]
On Ubuntu, when I enter `pip3 install -r Requirements.txt`, I get the output:
`ERROR: Could not open requirements file: [Errno 2] No such file or directory: 'Requirements.txt'`

**Comment by somebadcode** [comment]
The filename is `requirements.txt`, it's all lower case.

**Reply by NetworkJayJay007** [reply]
Ah, yes, I've tried it both ways and the output is:
`ERROR: Could not open requirements file: [Errno 2] No such file or directory: 'requirements.txt'`
Everything before worked fine.

**Reply by somebadcode** [reply]
Are you sure that you're running the command while inside the root of the repository?

What do the following commands say?

1.  `git rev-parse --show-toplevel`
2.  `pwd`

**Reply by NetworkJayJay007** [reply]
1. I get the output: `fatal: not a git repository (or any of the parent directories): .git`
2. My pwd is `/home/bob`

However, I also used the command `sudo pip3 install -r requirements.txt` and it still outputted the same thing.

When my pwd is `/` and enter `pip3 install -r requirements.txt`, it gives me the same output.
Same with if I enter `sudo su` to be as root, and enter the same command, same output of `ERROR: Could not open requirements file: [Errno 2] No such file or directory: 'requirements.txt'`

**Reply by somebadcode** [reply]
Have you cloned the repository?
Also, don't run `sudo`. Use a virtual environment instead. When you use virtual environments, the Python project will have its own dependencies separate from the rest of the system and other projects. Projects don't always support the same versions of packages so they need to have their own versions installed.

```shell
mkdir ~/git
cd ~/git
git clone https://github.com/ThioJoe/YouTube-Spammer-Purge.git
cd YouTube-Spammer-Purge
python3 -m venv venv
source venv/bin/activate
pip install --upgrade pip wheel setuptools
pip install -r requirements.txt
pip install requests
```
The last line is because `requests` is missing from `requirements.txt` but most users won't have an issue with it since there's probably already a system wide installation of it, which is probably why it's not in there. Everything that isn't part of Python's standard library must be listed in `requirements.txt`.

Every time you come back and want to run or work on the script, you need to run `source venv/bin/activate` while in the root of the cloned repository to make the packages available to the script. When you're done you either close the terminal or you run `deactivate` to reverse what the sourcing of the activate script did.
